server/chat-plugins/rock-paper-scissors.ts

```diff
@@ -0,0 +1,426 @@
+				);
+			}
+			const existingRoom = findExisting(user.id, targetUser.id);
+			const roomid = existingRoom ? existingRoom.roomid : `rps-${user.id}-${targetUser.id}`;
+			if (existingRoom?.game && !existingRoom.game.ended) {
+				return this.errorReply(`You already have a Rock, Paper, Scissors game against ${targetUser.name}.`);
+			}
+			if (!this.pmTarget) this.pmTarget = targetUser;
+			challengeRequests.set(targetUser.id, [user.id, rounds]);
+			this.sendChatMessage(
+				`/raw ${user.name} challenged you to Rock, Paper, Scissors!` +
+				`<button class="button" name="send" value="/j ${roomid}"><strong>Accept</strong></button></div>`,
+			);
+		},
+
+		accept(target, room, user) {
+			const request = challengeRequests.get(user.id);
+			if (!request) return this.errorReply(`You have no Rock, Paper, Scissors request pending from ${target}.`);
+			const [id, rounds] = request;
+			const targetUser = Users.get(id)!;
+			const existingRoom = findExisting(user.id, targetUser.id);
+			const options = {
+				modchat: '+',
+				isPrivate: true,
+			};
+			const roomid = `rps-${targetUser.id}-${user.id}`;
+			const gameRoom = existingRoom ? existingRoom : Rooms.createGameRoom(
+				roomid as RoomID, `[RPS] ${user.name} vs ${targetUser.name}`, options
+			);
+			gameRoom.game = new RPSGame(gameRoom, parseInt(rounds));
+			gameRoom.add(
+				`|raw|<h2>Welcome to the Rock, Paper, Scissors game of ${user.name} vs ${targetUser.name}!</h2>` +
+				`Use /rps start to start the game, once both players have joined!`
+			).update();
+			user.joinRoom(gameRoom.roomid);
+			targetUser.joinRoom(gameRoom.roomid);
+		},
+
+		deny(target, room, user) {
+			const request = challengeRequests.get(user.id);
+			if (!request) return this.errorReply(`You have no Rock, Paper, Scissors challenge pending.`);
+			const [sender] = request;
+			Users.get(sender)?.popup(`${user.name} denied your Rock, Paper, Scissors challenge.`);
+			challengeRequests.delete(user.id);
+		},
+
+		end(target, room, user) {
+			const game = getGame(this);
+			if (!game.playerTable[user.id]) {
+				return this.errorReply(`You are not a player, and so cannot end the game.`);
+			}
+			game.end(true);
+		},
+
+		choose(target, room, user) {
+			this.parse(`/choose ${target}`);
+		},
+
+		leave(target, room, user) {
+			this.parse(`/leavegame`);
+		},
+
+		start(target, room, user) {
+			const game = getGame(this);
+			if (game.roundTimer) return this.errorReply(`The game has already started.`);
+			game.start();
+		},
+
+		pause(target, room, user) {
+			const game = getGame(this);
+			game.pause(user);
+		},
+
+		resume(target, room, user) {
+			const game = getGame(this);
+			game.unpause(user);
+		},
+
+		help() {
+			this.runBroadcast();
+			const strings = [
+				`/rockpaperscissors OR /rps<br />`,
+				`/rps create OR /rps challenge [user], [rounds] - Challenges a user to a game of Rock, Paper, Scissors with [rounds] (or 3) rounds`,				`/rps leave - Leave the game.`,
```

Comment:
I think by default it should just be infinite rounds (stop whenever you want to stop). Might be worth asking, though.

Also I don't think you should need to specify the user to specify the number of rounds.

Reply:
Honestly yeah, ~infinite might be better. It's pretty short otherwise. I'll enforce a max of ~500 (?), though. (not like they're super weighty)